### PR TITLE
G5 Linked_list

### PR DESCRIPTION
**Pull request by eliloz**
**Team Members:**
- Prashant
- Eli

### CODE REVIEW

File: linked_list/main.py [hunk 1]
```diff
@@ -44,16 +45,20 @@ class Node(object):
...
 
     def __eq__(self, other):
-        pass
+        if self.elem == other.elem:
+            return True
+        return False
 
     def __repr__(self):
-        pass
+        return "Element: {}, next = {}".format(self.elem, self.next)
```
Comment (ayellapragada) [inline linked_list/main.py:61]: While this works, I think you should change Element to Node.

While its easy to tell whats what in this case, if you were working on a more complicated project, multiple objects could all have an Element section, if you're returning the repr(object), you want to know that it's a node.

File: linked_list/main.py [hunk 2]
```diff
@@ -62,34 +67,72 @@ class LinkedList(AbstractLinkedList):
...
     def __init__(self, elements=None):
-        pass
-
+        self.start = None
+        self.end = None
+        self.length= 0 
+        if elements:
+            for elem in elements:
+                self.append(elem)
+
```
Comment (ayellapragada) [inline linked_list/main.py:76]: could you not set self.elements = to elements ?

File: linked_list/main.py [hunk 3]
```diff
@@ -62,34 +67,72 @@ class LinkedList(AbstractLinkedList):
...
+            for i in range(index):
+                target = target.next
+        return target
 
     def __add__(self, other):
-        pass
+        for elem in other:
+           self.append(elem)
+        return self
+
```
Comment (ayellapragada) [inline linked_list/main.py:99]: smart way of doing add and iadd

i struggled with this and heavily relied on lists, this is a much better way than what i did

File: linked_list/main.py [hunk 4]
```diff
@@ -62,34 +67,72 @@ class LinkedList(AbstractLinkedList):
...
     def __iadd__(self, other):
-        pass
+        self.append(other)
+        return self
 
     def __eq__(self, other):
-        pass
-
+        return self.start == other.start and self.end == other.end and self.length ==other.length
+
```
Comment (ayellapragada) [inline linked_list/main.py:107]: I feel like I'm misunderstanding how your objects work, but is this only checked the first and last elements of your list, and the length?

**Comment by jon003** [comment]
A lot of your test cases are failing.  run 'tox' in your base directory and you can see and correct the errors.

**Pull request closed without merging**